meta-balena-common/recipes-support/resin-mounts/resin-mounts.inc

```diff
@@ -55,6 +55,8 @@ do_compile () {
 			# This bind mount is only invoked manually in persistent logs
 			# service. No need to run it with default target.
 			sed -i -e "/^WantedBy=/ d" $servicefile
+			# Mount persistent logs on data partition
+			sed -i -e 's#/mnt/state/root-overlay#/mnt/data/resin-data/logs#g' $servicefile
```

Comment:
Also need to `sed` the `resin-state.service` mention in `Before=`/`After=` lines in the template.
I'd recommend `sed` on the whole word i.e. `resin-state.service` and not just `resin-state`

Also lets start adding double quotes to `"$servicefile"`. At least in newer additions.